src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointTimeline.tsx

```diff
@@ -28,19 +30,24 @@ function Points({
   hoveredItem: HoveredItem | null;
 }) {
   const executionPoint = useSelector(getExecutionPoint);
-  let displayedPoints = [...analysisPoints.data];
-
-  // Even if we're over maxHitsDisplayed, we should still display a single paused
-  // point if the user happens to be paused on that this breakpoint.
-  if (analysisPoints.data.length > prefs.maxHitsDisplayed && executionPoint) {
-    displayedPoints = displayedPoints.filter(
-      point => BigInt(point.point) == BigInt(executionPoint)
-    );
-  }
+  const displayedPoints = useMemo(() => {
+    let previousDisplayed: PointDescription;
+    return analysisPoints.data.filter(p => {
+      if (
+        !previousDisplayed ||
+        executionPoint === p.point ||
+        p.time - previousDisplayed.time > HALF_A_SECOND
```

Comment:
Need to double check but isn't time here in ms, so `HALF_A_SECOND` should be 500?

Reply:
Yeah I was also confused about this, but after double-checking I think A) you are correct and B) we should probably have it be larger than 5ms but also smaller than 500. I think I'll play with it a little and decide what feels right.

Reply:
What I've realized should have been obvious - it depends on how long the recording is. I've made it 1% of the recording, so the dots are thick enough that they can run together, but bursty activity gets sampled down to max 100 events.